lib/typecheck.js

```diff
@@ -771,8 +777,33 @@ async function typeCheckPermissionRule(permissionRule, schemas, getMeta = false)
     }
 }
 
+async function typeCheckDataset(dataset, schemas, classes, getMeta = false) {
+    for (let ex of dataset.examples) {
+        await typeCheckDeclaration(ex, schemas, new Scope(), classes, getMeta, true);
+        if (!Array.isArray(ex.utterances))
+            throw new TypeError('Utterances annotation expects an array');
+        for (let utterance of ex.utterances) {
+            if (!utterance.isString)
+                throw new TypeError('Utterance can only be a string');
```

Comment:
Same here, is utterance a Ast.Value.String or a String?

Reply:
fixed.